File: Projects/Project_02/For_Students/matlab/time_integration.m
%---------------------------------------------------------------------%
%This function performs the time-integration
%Written by F.X. Giraldo on April 19, 2019
%           Department of Applied Mathematics
%           Naval Postgraduate School
%           Monterey; CA 93943-5216
%---------------------------------------------------------------------%
function [q0,time] = time_integration(q0,Dhat,periodicity,time,ntime,dt)
    
[q0,time] = ti_LSRK(q0,Dhat,periodicity,time,ntime,dt);

File: Projects/Project_02_1D_Wave/For_Students/julia/time_integration.jl
#---------------------------------------------------------------------#
#This function performs the time-integration
#Written by F.X. Giraldo on April 19, 2019
#           Department of Applied Mathematics
#           Naval Postgraduate School
#           Monterey; CA 93943-5216
#---------------------------------------------------------------------#
include("ti_SSP.jl")
include("ti_LSRK.jl")

function time_integration!(q0,u,Dhat,Fhat,intma,periodicity,time,ntime,dt,stages,DFloat,ti_type)

    if (ti_type == "SSP")
        q0 = ti_SSP!(q0,u,Dhat,Fhat,intma,periodicity,time,ntime,dt,stages,DFloat)
    elseif (ti_type == "LSRK")
        q0 = ti_LSRK!(q0,u,Dhat,Fhat,intma,periodicity,time,ntime,dt,DFloat)
    end
    return (q0)
end
